### PR TITLE
MailChimp dashboard UI view.

### PR DESCRIPTION
**Pull request by budzanowski**
### Information

This is a part of #17434 PR that represents MailChimp Dashboard UI view. This is what user sees when MailChimp is ready and configured.
This is provided for review as a separate PR because the original (spike) PR is too big for the review to be comfortable. 

To test this component in action please refer to #17434.

#### Warning

In the final version, the button to start setup wizard will be removed. The design does include it in this place/stage.

The card needs some styling. 

image

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+			<span className="mailchimp__dashboard-settings-form">
+				<FormFieldset>
+					<FormLegend>{ translate( 'Newsletter subscriptions' ) }</FormLegend>
+					<FormLabel>
+						<FormToggle
+							checked={ toggle }
+							onChange={ onToggle }
+							id="show-subscribe-message"
+						/>
+						<span>{ translate( 'Show a subscribe message to ustomer at checkout' ) }</span>
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: `ustomer` -> `customers`

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+						<FormRadio
+							disabled={ ! toggle }
+							value="uncheck"
+							checked={ 'uncheck' === checkbox }
+							onChange={ onRadioChange }
+						/>
+						<span>{ translate( 'Subscribe message is unchecked by default' ) }</span>
+					</FormLabel>
+					<FormLabel>
+						{ translate( 'Subscribe message is unchecked by default' ) }
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: This string is repeated here again for the newsletter label. It should just be

'Subscribe Message'

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	);
+
+	const onResyncCLick = () => {
+		resync( siteId );
+	};
+
+	return (
+		<div>
+			<div>
+				<span className="mailchimp__account-info">{ translate( 'MailChimp account:' ) }</span>
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: I think we need to have a loading indicator here so we can show something before the mailing list info is ready. I get a flash of 'undefined' before the actual list name displays.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+	const products = hasProductInfo ? ( product_count + '/' + mailchimp_total_products ) : 'X/X';
+	const hasOrdersInfo = ( order_count !== undefined ) && ( mailchimp_total_orders !== undefined );
+	const orders = hasOrdersInfo ? ( order_count + '/' + mailchimp_total_orders ) : 'X/X';
+
+	const synced = () => (
+		<Notice
+			status="is-success"
+			isCompact={ true }
+			showDismiss={ false }
+			text={ syncState.mailchimp_list_name + ' ' + translate( 'list synced' ) }>
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: This should be something like
```
translate( '%(mailingListname)s list synced.', {
				args: { mailingListname: syncState.mailchimp_list_name },
} );
```

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+		</Notice>
+	);
+
+	const syncing = () => (
+		<Notice
+			status="is-warning"
+			isCompact={ true }
+			showDismiss={ false }
+			text={ syncState.mailchimp_list_name + ' ' + translate( 'list is being synced' ) }>
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: See above comment on line 43.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+import { isSubmittingNewsletterSetting, newsletterSettingsSubmitError } from 'woocommerce/state/sites/settings/email/selectors';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+const SyncTab = localize( ( { siteId, translate, syncState, resync } ) => {
+	const { account_name, store_syncing, product_count, mailchimp_total_products,
+		mailchimp_total_orders, order_count } = syncState;
+	const hasProductInfo = ( product_count !== undefined ) && ( mailchimp_total_products !== undefined );
+	const products = hasProductInfo ? ( product_count + '/' + mailchimp_total_products ) : 'X/X';
+	const hasOrdersInfo = ( order_count !== undefined ) && ( mailchimp_total_orders !== undefined );
+	const orders = hasOrdersInfo ? ( order_count + '/' + mailchimp_total_orders ) : 'X/X';
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: If we can't get the counts, should we even bother displaying X/X?

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+import FormLegend from 'components/forms/form-legend';
+import FormToggle from 'components/forms/form-toggle';
+import FormRadio from 'components/forms/form-radio';
+import FormTextInput from 'components/forms/form-text-input';
+import Notice from 'components/notice';
+import QueryMailChimpSyncStatus from 'woocommerce/state/sites/settings/email/querySyncStatus';
+import {
+	syncStatus,
+	mailchimpSettings,
+	isRequestingSettings } from 'woocommerce/state/sites/settings/email/selectors';
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: `} from 'woocommerce/state/sites/settings/email/selectors';`

should go on the next line.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+		super( props );
+		this.state = {
+			syncStatus: null,
+			settings: props.settings
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { translate } = nextProps;
+		if ( ( nextProps.isSubmittingNewsletterSetting === false ) && this.props.isSubmittingNewsletterSetting ) {
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: Yoda `false === nextProps.isSubmittingNewsletterSetting`

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+		mailchimp_total_orders, order_count } = syncState;
+	const hasProductInfo = ( product_count !== undefined ) && ( mailchimp_total_products !== undefined );
+	const products = hasProductInfo ? ( product_count + '/' + mailchimp_total_products ) : 'X/X';
+	const hasOrdersInfo = ( order_count !== undefined ) && ( mailchimp_total_orders !== undefined );
+	const orders = hasOrdersInfo ? ( order_count + '/' + mailchimp_total_orders ) : 'X/X';
+
+	const synced = () => (
+		<Notice
+			status="is-success"
+			isCompact={ true }
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: I think you can just have `isCompact` here

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+			showDismiss={ false }
+			text={ syncState.mailchimp_list_name + ' ' + translate( 'list synced' ) }>
+		</Notice>
+	);
+
+	const syncing = () => (
+		<Notice
+			status="is-warning"
+			isCompact={ true }
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: Same as above comment.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+			showDismiss={ false }
+			text={ syncState.mailchimp_list_name + ' ' + translate( 'list is being synced' ) }>
+		</Notice>
+	);
+
+	const onResyncCLick = () => {
+		resync( siteId );
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: Should we check to see if another resync is already in progress?

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+	};
+
+	return (
+		<div>
+			<div>
+				<span className="mailchimp__account-info">{ translate( 'MailChimp account:' ) }</span>
+				<span>{ account_name }</span>
+			</div>
+			<span>{ store_syncing ? syncing() : synced() }</span>
+			<a onClick={ onResyncCLick }>Resync</a>
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: missing `translate()`

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+							checked={ 'uncheck' === checkbox }
+							onChange={ onRadioChange }
+						/>
+						<span>{ translate( 'Subscribe message is unchecked by default' ) }</span>
+					</FormLabel>
+					<FormLabel>
+						{ translate( 'Subscribe message is unchecked by default' ) }
+					</FormLabel>
+					<FormTextInput
+						name={ 'newsletter_label' }
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: no need for the `{}` here.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+						onChange={ onNewslatterLabelChange }
+						value={ settings.newsletter_label }
+					/>
+				</FormFieldset>
+			</span>
+			<span className="mailchimp__dashboard-settings-preview">
+				<div>{ translate( 'PREVIEW' ) }</div>
+				<div className="mailchimp__dashboard-settings-preview-view">
+					{toggle && <FormLabel>
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: missing space `{ toggle`

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+	);
+} );
+
+class MailChimpDashboard extends React.Component {
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			syncStatus: null,
+			settings: props.settings
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: Trailing comma

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 16]
```diff
@@ -0,0 +1,249 @@
...
+			mailchimp_auto_subscribe: settings.mailchimp_auto_subscribe,
+			mailchimp_checkbox_defaults: settings.mailchimp_checkbox_defaults,
+			mailchimp_checkbox_action: settings.mailchimp_checkbox_action,
+		};
+		submit( siteId, message );
+	}
+
+	render() {
+		const { siteId, syncStatusData, translate } = this.props;
+		const slogan = translate( 'Allow customers to subscribe to your MailChimp email list' );
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: We could just have this translate call inline on line 206

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 17]
```diff
@@ -0,0 +1,249 @@
...
+
+		return (
+			<div>
+				<QueryMailChimpSyncStatus siteId={ siteId } />
+				<Card className="mailchimp__dashboard" >
+					<div className="mailchimp__dashboard-first-section" >
+						<span className="mailchimp__dashboard-title-and-slogan">
+							<div className="mailchimp__dashboard-title">Mailchimp</div>
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: `capital_C_dangit()`

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 18]
```diff
@@ -0,0 +1,249 @@
...
+							<div>{ slogan }</div>
+						</span>
+						<span className="mailchimp__dashboard-sync-status" >
+							<SyncTab
+								siteId={ siteId }
+								syncState={ syncStatusData }
+								resync={ this.props.requestResync } />
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: could add `requestResync` to the props assignment on line 196. Few more variables you could do the same with below.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js (hunk 6 above)
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: hmm, true empty string will actually be better.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js (hunk 18 above)
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: True but it creates name clashes with imports. What would be the best strategy here. Have a different name in connect or extract to a different name like { requestResync: reqResync } = ...
That is why I have originally left it this way.

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js (hunk 3 above)
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: @kellychoffman ^ how would you like me to tackle this?

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js (hunk 18 above)
Comment (justinshreve) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: When I run into naming clashes like that, I just continue using `this.props.whatever` like you have.

Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: Most of the things tackle. Ready for more bashing :)

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js (hunk 3 above)
Comment (justinshreve) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: If you detect that the info is still loading, you can instead show a placeholder div.
That div can include some CSS that will show a pulsing empty div like we do in other places of the app. Example: https://github.com/Automattic/wp-calypso/blob/782eb8610aca534ec92aaf035b7cbfa34cd9d4c3/client/extensions/woocommerce/components/store-address/style.scss#L1-L5

File: client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js [hunk 19]
```diff
@@ -0,0 +1,300 @@
...
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { translate } = nextProps;
+		if ( ( false === nextProps.isSaving ) && this.props.isSaving ) {
+			if ( nextProps.newsletterSettingsSubmitError ) {
+				nextProps.errorNotice( translate( 'There was a problem saving the email settings. Please try again.' ) );
+			} else {
+				nextProps.successNotice( translate( 'Email settings saved.' ), /*{ duration: 4000 }*/ );
```
[the remark this answers is not in the document]
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js]: OH NO! we want duration here I have commented it just for some testing purposes.

**Pull request merged**